src/main/java/com/minecolonies/coremod/client/gui/WindowHutWareHouse.java

```diff
@@ -247,6 +315,151 @@ public void updateElement(final int index, @NotNull final Pane rowPane)
         });
     }
 
+    /**
+     * Update the item list.
+     */
+    private void updateResources()
+    {
+        List<ItemStorage> filterItems = new ArrayList<>();
+        final List<BlockPos> containerList = building.getContainerList();
+        List<ItemStack> items = new ArrayList<>();
+        int count = containerList.size();
+        World world = building.getColony().getWorld();
+
+        for (int s = 0; s < count; s++)
+        {
+            final TileEntity rack = world.getTileEntity(containerList.get(s));
```

Comment:
Btw, there is already a way to get ItemStorage from the rack without having to go through all slots

Reply:
I need to go through all slots of all Racks because i am summing all up. because Multiple Stacks of the same type can be in multiple Racks. This function creates basically a single ItemStack List with all combined.

The getContainerlist returns only the block position from the building. i couldnt access it directly because i am in the View part. so i need to serialize and deserialize it. i serialized the containerList from the building side and deserialized it in the view part. from there i get the inventory of each rack. and then i need to sum up all together. then adding the filter and set it to the allItems List and call the updateResourceList